packages/analytics-js/src/components/core/Analytics.ts

```diff
@@ -547,7 +561,11 @@ class Analytics implements IAnalytics {
       return;
     }
 
-    this.userSessionManager?.setGroupId(payload.groupId);
+    // `null` value indicates that previous group ID needs to be retained
```

Comment:
again, how can you clear it if that is the case?

Reply:
Similar to the user ID case above.
https://github.com/rudderlabs/rudder-sdk-js/blob/723955d6f408cf97ec7e1a17c4467abcdb2bff10/src/core/analytics.js#L778